src/cli/config/init.ts

```diff
@@ -24,12 +24,28 @@ export const run: CliCommand = async (args: Arguments /* , logger: Logger */) =>
   const pkgFile = isPackage ? filePath : join(process.cwd(), 'package.json')
   const hasPackage = isPackage || existsSync(pkgFile)
   // read config
-  const { allowJs = false, jestPreset = true, tsconfig: askedTsconfig, babel = false, force, jsdom } = args
+  const { jestPreset = true, tsconfig: askedTsconfig, force, jsdom } = args
   const tsconfig = askedTsconfig === 'tsconfig.json' ? undefined : askedTsconfig
-  const hasPresetVar = allowJs || !jestPreset
   // read package
   const pkgJson = hasPackage ? JSON.parse(readFileSync(pkgFile, 'utf8')) : {}
 
+  // auto js/babel
+  let { js, babel } = args
+  if (js != null || babel != null) {
```

Comment:
what is going on with these ternaries.
if js is null, we set it to either 'babel or undefined'
if babel is null, we set it to.. true or false depending on the value of js?

This is really hard to decipher. Can we rewrite this?

Reply:
No, because I don't understand what it's doing and cannot remember... ROFLMAO

yeah, definitely needs a clearer version, will do now

Reply:
Would this be ok?

```ts
  // auto js/babel
  let { js: jsFilesProcessor, babel: shouldPostProcessWithBabel } = args
  // set defaults for missing options
  if (jsFilesProcessor != null || shouldPostProcessWithBabel != null) {
    if (jsFilesProcessor == null) {
      // set default js files processor depending on whether the user wants to post-process with babel
      jsFilesProcessor = shouldPostProcessWithBabel ? 'babel' : undefined
    } else if (shouldPostProcessWithBabel == null) {
      // auto enables babel post-processing if the user wants babel to process js files
      shouldPostProcessWithBabel = jsFilesProcessor === 'babel'
    }
  }
```

Reply:
Hah, I know that feeling.
Are both of them ever enabled at the same time? E.g. does it happen that both jsFilesProcessor and and shouldPOstProcessWithBabel both are null? Because otherwise we don't need the first if statement.

Actually I think no matter what we don't need it, because the assignments inside are null-safe.

Reply:
They could be both null or both set, or only one of each set. They are args coming from the user input at CLI:

- `ts-jest config:init`
- `ts-jest config:init --js babel` // js files are processed with babel, this will auto set --babel to true
- `ts-jest config:init --js babel --no-babel` // js files processed with babel, but ts-jest will not post process ts files with babel
- `ts-jest config:init --babel` // same as `--js babel`
- `ts-jest config:init --js ts` // js files will also be compiled with ts-jest using typescript
- `ts-jest config:init --js ts --babel` // ts-jest will process js files with typescript, and post-process everything with babel

Reply:
btw, `ts-jest help config:init` describes those options ;-)

Reply:
I still think we can scrap the ` if (jsFilesProcessor != null || shouldPostProcessWithBabel != null) `

All assignments inside work even if the opposite part is undefined.

Reply:
You're right, here it doesn't matter because of the overriding to `undefined`, which at the end is like null.

But in the strict World, or if we want to change this `undefined` with a truthy value, the result of the 2 are different. In your version it'll reset `jsFilesProcessor` even if both are unset while here it should reset it only if any of the 2 are set.

What I meant here is "IF and ONLY IF the user defined either one or the other, process with setting up defaults, whatever those defaults are and whatever logic it is".

I'll update and remove the surrounding if.